src/targets/furnace.js

```diff
@@ -0,0 +1,98 @@
+import { ClickableObject } from "../clickable-object.js";
+import { CONSTANTS } from "../constants/constants.js";
+import { characterData } from "../cookie-io.js";
+import { Button } from "../ui/button.js";
+import { getItemClass } from "../utilities.js";
+
+export class Furnace extends ClickableObject {
+    name = "Furnace";
+    varName = "bar"; // Use # bars forged as quest
+    examineText = "A red hot furnace.";
+    actions = [
+        { text: "Forge", func: "clickTarget" },
+        { text: "Examine", func: "examine" },
+    ];
+
+    constructor(scene) {
+        super();
+        this.scene = scene;
+
+        const cameraWidth = scene.cameras.main.width;
+        const cameraHeight = scene.cameras.main.height;
+        const x = cameraWidth / 2 - 230;
+        const y = cameraHeight / 2 - 150;
+        const width = 230;
+        const height = 150;
+
+        // Add invisible button for furnace
+        this.sprite = new Button(scene, x, y, width, height);
+        this.sprite.on("pointerdown", (pointer) => {
+            if (pointer.rightButtonDown()) {
+                this.createRightClickMenu(pointer.x, pointer.y, this.actions);
+            } else {
+                this.clickTarget();
+            }
+        });
+    }
+
+    async clickTarget() {
+        const inv = this.scene.dashboard.inventory;
+        const chat = this.scene.scene.get(CONSTANTS.SCENES.CHAT);
+
+        // See if ore is selected
+        const selectedIndex = inv.curSelectedItemIndex;
+        if (selectedIndex < 0) {
+            chat.writeText("Select an ore in your inventory first");
+            return;
+        }
+
+        const selectedItem = inv.inventory[selectedIndex];
+        switch (selectedItem.name) {
+            case "Copper Ore":
+            case "Tin Ore":
+                this.smelt(
+                    ["CopperOre", "TinOre"],
+                    "BronzeBar",
+                    1,
```

Comment:
RS gives 6.25 xp for bronze bars

Reply:
xp should also probably be defined in the class. You can remove the parameter and get xp from the item in the smelt method

Reply:
1) I tried that first but I leveled up so quickly. I guess that doesn't take into account time taken to get ore though...
2) That made sense for bones since you bury them directly but I'm not sure if it makes sense here, xp isn't stored in the ore or logs, it's stored in the rock/tree object which is most similar to Furnace here

Reply:
Updated xp to 6. You haven't convinced me on putting the xp in the bronze bar class yet though

Reply:
I think the xp should be stored somewhere so you can reference it as the xp gained from smelting a bronze bar. Storing in the class would encapsulate information that one should not need to know when calling the smelt method. It reduces the complexity overall of the smelt method and this case statement. Calling smelt("BronzeBar") is so much cleaner.

Reply:
Done https://github.com/EthanVieira/rs-clicker/pull/60/commits/d0e349d408a86d7b07a378c05bf957c70fb030cc